drivers/serial/usart_gd32.c

```diff
@@ -93,25 +148,201 @@ static int usart_gd32_err_check(const struct device *dev)
 	return errors;
 }
 
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+/** Interrupt driven FIFO fill function */
```

Comment:
Remove all useless comments

Reply:
Remove the API comments which copy from drivers/uart.h